cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunnerTest.java

```diff
@@ -312,9 +314,18 @@ public void testMapReduceDriverResources() throws Exception {
 
   @Test
   public void testMapreduceWithObjectStore() throws Exception {
-    final ApplicationWithPrograms app = deployApp(AppWithMapReduceUsingObjectStore.class);
+    // Deploy apps to another namespace and test cross-namespace access meanwhile
+    final ApplicationWithPrograms app = deployApp(NamespaceId.fromString("namespace:someOtherNameSpace").toId(),
+                                                  AppWithMapReduceUsingObjectStore.class);
 
-    final ObjectStore<String> input = datasetCache.getDataset("keys");
+    final ObjectStore<String> input = datasetCache.getDataset("someOtherNameSpace", "keys");
+
+    // Get dataset from a non existing namespace
+    try {
+      final ObjectStore<String> inputwrong = datasetCache.getDataset("nonExistingNameSpace", "keys");
```

Comment:
put Assert.fail("msg") after line 325. Otherwise, the test case will still pass, even if the getDataset() call throws no exception.